test/data/kms-tests.json

```diff
@@ -124,5 +124,79 @@
       "\r\n"
     ],
     "expect": "Could not parse Status-Code"
+  },
+  {
+    "description": "Successful oauth request",
+    "ctx": ["azure_oauth_datakey", "azure_oauth_decrypt", "gcp_oauth_datakey", "gcp_oauth_decrypt"],
```

Comment:
This tests both datakey and decryption, since they go through different code paths for creating KMS requests (encryption/decryption goes through mongocrypt-key-broker.c, creating a datakey in mongocrypt-ctx-datakey.c).